File: src/buildings/storage.js
import * as PIXI from 'pixi.js'
import { Building } from '../building.js'

export class Storage extends Building {

    constructor (game) {
        super("Storage",{x:2,y:2},[[0,0],[0,0]],game,false)
        this.sprite = PIXI.Sprite.from('Storage')

        this.capacity = 20

        this.description = `A container that can hold up to ${this.getCapacity()} items.`
    }

    getCapacity() {
        return this.capacity
    }
}

File: TriloGame.CSharp/src/TriloGame.Game/Core/Buildings/Storage.cs
using TriloGame.Game.Core.Simulation;
using TriloGame.Game.Shared.Math;

namespace TriloGame.Game.Core.Buildings;

public sealed class Storage : Building
{
    public Storage(GameSession session)
        : base("Storage", new GridPoint(2, 2), [[0, 0], [0, 0]], session, false)
    {
        TextureKey = "Storage";
        Recipe = new Dictionary<string, int>(StringComparer.Ordinal) { ["Sandstone"] = 20 };
        Capacity = 20;
        Description = $"A container that can hold up to {Capacity} items.";
    }

    public int Capacity { get; }
}
